### DISCUSSION TITLE
Allow docstrings on tests

### DISCUSSION THREAD

**Discussion by jariji** [Seelengrab/Supposition.jl#55] [Design & Ideas]
I like to put docstrings on my tests so I know what they are testing. 

```jl
"abs returns a nonnegative value."
@check function abs_nonneg(x=Floats())
  zero(x) <= abs(x)
end

ERROR: cannot document the following expression
```

**Comment by Seelengrab** [comment]
I don't think there's anything I can do to make that syntax work; you just can't put a docstring on a `begin` block expression (which is what's returned by `@check`) like that. However, since `abs_nonneg` is a function like any other, you can document it with `@doc`, or by documenting a function prototype:

```julia
"abs returns a nonnegative value."
function abs_nonneg end
```
